### PR TITLE
Issue 25156 supporting more complex parameters type

### PR DESCRIPTION
**Pull request by freddyucv**
This is the first draft PR of https://github.com/dotCMS/core/issues/25156

In this issue we need to create a new Metric Type, each Metric Type can allow a set of Parameter and these parameters are check later if they meet a Condition when they are use inside a Metric, for this Condition we support a set of Operator

https://github.com/dotCMS/core/blob/14f9a850bb545d5b8e0d709523cb8e10be4e8a81/dotCMS/src/main/java/com/dotcms/cube/filters/SimpleFilter.java#L80

We check if the Condition is meet here

https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotcms/analytics/metrics/AbstractMetric.java#L44-L51

We take the value to check the Operator from the Event for now what we do if just take a Event's attribute value with the same name of the Parameter but for the new MetricType we need to support something more complex.

### Proposed Changes
*  We need to support different type of Parameter each one with its own Transformer to transform its value, it is going to allow us support Parameter that is configuration values are not just a Simple String.

https://github.com/dotCMS/core/pull/25507/files#diff-75510bc7dededdfed8b986435f8be1f23e53e726a1f3e3e79e9b2b48e8930cd6R86

https://github.com/dotCMS/core/pull/25507/files#diff-b75abee68c401787eea2b53d0444adf818b5fb9bb0c6f996714bed212823fb26R9

* How I mention before we need to support more complex way to extract values from the Events, Also we need to handle a set of Values

https://github.com/dotCMS/core/pull/25507/files#diff-ff8fe68affeba157cb5f8967535772b80d1b77b9e0fa167f12dac1594370e21dR10 

https://github.com/dotCMS/core/pull/25507/files#diff-75510bc7dededdfed8b986435f8be1f23e53e726a1f3e3e79e9b2b48e8930cd6R41

This first PR is just a refactoring so I don't need create any test

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/analytics/metrics/AbstractCondition.java [hunk 1]
```diff
@@ -22,6 +27,22 @@ public interface AbstractCondition {
     Operator operator();
     String value();
 
+    @JsonIgnore
+    default boolean isValid(final MetricType type, final Condition condition, final Event event){
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/analytics/metrics/AbstractCondition.java:31]: doc

File: dotCMS/src/main/java/com/dotcms/analytics/metrics/DefaultParameterValueGetter.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.dotcms.analytics.metrics;
+
+import com.dotcms.analytics.metrics.AbstractCondition.AbstractParameter;
+import com.dotcms.experiments.business.result.Event;
+
+
+public class DefaultParameterValueGetter implements ParameterValueGetter<String> {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/analytics/metrics/DefaultParameterValueGetter.java:7]: doc

File: dotCMS/src/main/java/com/dotcms/analytics/metrics/ParameterValueGetter.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.dotcms.analytics.metrics;
+
+import com.dotcms.experiments.business.result.Event;
+
+public interface ParameterValueGetter<T> {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/analytics/metrics/ParameterValueGetter.java:5]: doc

File: dotCMS/src/main/java/com/dotcms/analytics/metrics/AbstractCondition.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/analytics/metrics/AbstractCondition.java]: done https://github.com/dotCMS/core/pull/25507/files#diff-75510bc7dededdfed8b986435f8be1f23e53e726a1f3e3e79e9b2b48e8930cd6R39

File: dotCMS/src/main/java/com/dotcms/analytics/metrics/DefaultParameterValueGetter.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/analytics/metrics/DefaultParameterValueGetter.java]: done https://github.com/dotCMS/core/pull/25507/files#diff-5d7494f5f79c4d5c3b700eef9440e8835700c69dd76647e33802986039733036R6

File: dotCMS/src/main/java/com/dotcms/analytics/metrics/ParameterValueGetter.java (hunk 3 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/analytics/metrics/ParameterValueGetter.java:9]: done https://github.com/dotCMS/core/pull/25507/files#diff-ff8fe68affeba157cb5f8967535772b80d1b77b9e0fa167f12dac1594370e21dR5

**Review by jdotcms — APPROVED** [review]

**Review by AndreyDotcms — APPROVED** [review]

**Pull request merged**